sql/core/src/test/scala/org/apache/spark/sql/ColumnExpressionSuite.scala

```diff
@@ -2790,19 +2788,20 @@ class ColumnExpressionSuite extends QueryTest with SharedSparkSession {
           ).foreach { case (end, start) =>
             val df = Seq((end, start)).toDF("end", "start")
             val daysBetween = Duration.ofDays(ChronoUnit.DAYS.between(start, end))
-            checkAnswer(df.select($"end" - $"start"), Row(daysBetween))
+            val t1= df.select($"end" - $"start")
```

Comment:
@cloud-fan
schema === [( end - start) : interval day]  DayTimeIntervalType(0, 0)
But the column value is DayTimeInterval( DAY, SECOND)   Default.

The value is store as Long, But it task the Long value of DayTimeIntervalType(0, 0) as Long value of DayTimeIntervalType(0, 3). How can I solve this problem? Can you give me some advice? 

When GetLong of LongColumnAccessor to convert the value ?